source/Weather.h

```diff
@@ -50,7 +50,7 @@ class Weather {
 	// The origin of the hazard.
 	const Point &Origin() const;
 	// Create any environmental effects and decrease the lifetime of this weather.
-	void Step(std::vector<Visual> &newVisuals);
+	void Step(std::vector<Visual> &newVisuals, Point flagship);
```

Comment:
Probably call it something other than `flagship`, that indicates it is an override of sorts.

(This is just stylistic, and prevents searching for "flagship" returning a result that is not a ship/pointer to ship.)